src/components/themeMediaPlayer.js

```diff
@@ -101,8 +103,25 @@ async function loadThemeMedia(serverId, itemId) {
 
         const result = userSettings.enableThemeVideos() && themeMedia.ThemeVideosResult?.Items?.length ? themeMedia.ThemeVideosResult : themeMedia.ThemeSongsResult;
 
+        let themeItems = result.Items;
+
+        if (userSettings.themeMediaSortBy() == ItemSortBy.Random) {
+            // Shuffle array using Durstenfeld algorithm https://web.archive.org/web/20240601200838/https://dl.acm.org/doi/pdf/10.1145/364520.364540#.pdf
+            const toShuffled = (array) => {
+                for (let i = array.length; i > 1; --i) {
+                    const lastIndex = i - 1;
+                    const randIndex = randomInt(0, lastIndex);
+                    // swap the elements in the randomly selected index and the last index
+                    [ array[randIndex], array[lastIndex] ] = [array[lastIndex], array[randIndex] ];
+                }
+            };
```

Comment:
I will switch to using the shuffle function from `lodash`.
I was looking the `apiclient.d.ts` to see if there was anyway kind of `SortItems` function on the `c#` side that accepted a `List<BaseItemDto>`, but I didn't find anything.

I agree a long term solution is to update the [getThemeMedia](https://github.com/jellyfin/jellyfin/blob/da07197c880b54ad054ae64efedff9add20bc0d9/Jellyfin.Api/Controllers/LibraryController.cs#L259-L271) function to support the SortBy and SortOrder options, so that the sorting happens on the server.
For now, I don't imagine it would be much of an issue as I don't think many users have a large number of audio or video files in their theme media folders.

Reply:
I have created a PR to update `GetThemeMedia` on the server side to support the `SortBy` and `SortOrder` properties: https://github.com/jellyfin/jellyfin/pull/12167